NeoMathEngine/src/GPU/Vulkan/VulkanMathEngine.cpp

```diff
@@ -284,51 +314,23 @@ CMemoryHandle CVulkanMathEngine::CopyFrom( const CMemoryHandle& handle, size_t s
 	return result;
 }
 
+
 CMemoryHandle CVulkanMathEngine::Alloc( size_t size )
 {
-	std::unique_ptr<CVulkanMemory> vulkanMemory( new CVulkanMemory );
-
-	VkBufferCreateInfo createInfo = {};
-	createInfo.sType = VK_STRUCTURE_TYPE_BUFFER_CREATE_INFO;
-	createInfo.size = size;
-	createInfo.usage = VK_BUFFER_USAGE_STORAGE_BUFFER_BIT | VK_BUFFER_USAGE_TRANSFER_SRC_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT;
-	createInfo.sharingMode = VK_SHARING_MODE_EXCLUSIVE;
-
-	VkBuffer buffer = 0;
-	vkSucceded( device->vkCreateBuffer( device->Handle, &createInfo, 0, &buffer ) );
-
-	VkMemoryRequirements req = {};
-	device->vkGetBufferMemoryRequirements( device->Handle, buffer, &req );
-
-	VkMemoryAllocateInfo allocInfo = {};
-	allocInfo.sType = VK_STRUCTURE_TYPE_MEMORY_ALLOCATE_INFO;
-	allocInfo.allocationSize = req.size;
-
-	// Look for the suitable index in memory
-	bool isFound = false;
-	const int flags = VK_MEMORY_PROPERTY_HOST_COHERENT_BIT | VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT;
-
-	for( int i = 0; i < (int)device->MemoryProperties.memoryTypeCount; ++i ) {
-		if((req.memoryTypeBits & (1 << i)) != 0 &&
-			(int)(device->MemoryProperties.memoryTypes[i].propertyFlags & flags) == flags) {
-			allocInfo.memoryTypeIndex = i;
-			isFound = true;
-			break;
-		}
+	try {
+		VkBufferUsageFlags usage = VK_BUFFER_USAGE_STORAGE_BUFFER_BIT | 
+			VK_BUFFER_USAGE_TRANSFER_SRC_BIT | 
+			VK_BUFFER_USAGE_TRANSFER_DST_BIT;
+		VkMemoryPropertyFlags properties = device->Type != VDT_Nvidia ?
```

Comment:
Let's move properties inside CVulkanMemory(). It's always the same.
And maybe 'usage' too...